resources/aws/gateway.go

```diff
@@ -114,7 +114,7 @@ func (g *Gateway) Delete() error {
 		}
 		return nil
 	}
-	if err := backoff.Retry(detachOperation, backoff.NewExponentialBackOff()); err != nil {
+	if err := backoff.Retry(detachOperation, NewCustomExponentialBackoff()); err != nil {
```

Comment:
What do you think about registering the retry callback and emit metrics for retry counts here?